typegate/tests/graphql/request_parser_test.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright Metatype OÜ under the Elastic License 2.0 (ELv2). See LICENSE.md for usage.
+
+import { FormDataParser } from "../../src/graphql/request_parser.ts";
+import { gql, test } from "../utils.ts";
+import * as mf from "test/mock_fetch";
+import { assertEquals, assertExists } from "std/testing/asserts.ts";
+
+mf.install();
+
+test("request parser: multipart/form-data", async (t) => {
+  let request: Request | null = null;
+  mf.mock("POST@/api/graphql", (req) => {
+    request = req;
+    return new Response("");
+  });
+
+  await t.should("successfully query with file", async () => {
+    const fileContent = "Hello, World!";
+    const q = gql`
+        mutation($file: File!) {
+          singleUpload(file: $file) { url }
+        }
+      `
+      .withVars({
```

Comment:
make withVars check the file type instead of have a withFile? we will likely publish the testing utilities as a separate package in a few weeks so we should make the easiest interface possible